### ISSUE TITLE
Passing redis configuration

### ISSUE THREAD

**Issue by Imunhatep** [docker-library/docs#1271]
From docs:

> Additionally, If you want to use your own redis.conf ...
> You can create your own Dockerfile that adds a redis.conf from the context into /data/, like so.
> FROM redis
> ...

This is a basic thing to do, and every one have to create own image for this? Seriously?

**Comment by wglambert** [comment]
The image by default runs with a default redis.conf. You can also pass config options as flags: `docker run --name some-redis -d redis redis-server --appendonly yes`, however any options not explicitly set will not be set.

You don't have to create your own image, it's just more efficient that way for continuous use. Just mounting the config is sufficient.

>Alternatively, you can specify something along the same lines with docker run options.
>`$ docker run -v /myredis/conf/redis.conf:/usr/local/etc/redis/redis.conf --name myredis redis redis-server /usr/local/etc/redis/redis.conf`

**Comment by Imunhatep** [comment]
Thank you, 

I have figured this out by inspecting dockerfile, but:
1. would be nice if docs had a note about it
2. it's not very convenient to keep configuration files with data in the same parent directory, at least for kubernetes installation
3. In the days of k8s, starting containers by passing cmd  is a crime

**Comment by tianon** [comment]
> would be nice if docs had a note about it

The docs do have a note about it; that's what @wglambert was quoting above.  See the _second_ half of "Additionally, If you want to use your own redis.conf ..." on https://hub.docker.com/_/redis/:

> Alternatively, you can specify something along the same lines with `docker run` options.

---

> it's not very convenient to keep configuration files with data in the same parent directory, at least for kubernetes installation

Not sure I get your meaning?  Your configuration file can live wherever you want both on your host/cluster and inside the image (you just need to specify the path to it when running `redis-server`), and your actual data can likewise be stored wherever you'd like (you just need to specify where with the `dir` Redis configuration option, which defaults to `./` or the current working directory).

---

> In the days of k8s, starting containers by passing cmd is a crime

Source?  The k8s pod YAML supports `command:` for this exact purpose (and is really no different than supplying an environment variable or even a configuration file in the realm of Kubernetes).

**Comment by Imunhatep** [comment]
1. Docs - sorry I missed this. I think I just stopped reading docs after I saw suggestion to creating own image for configuration..
2. Running several versions of redis and keeping it to use the same configs, leads us to use ConfigMap, instead of `command:`
3. I wanted to point out that this image stores redis data (by default) in the same dir where redis.conf is located..  But at second look I see it uses `/data` dir, and I can't find where it's configured.. Since default value (at least according to the docs) is `./` . 

My main concern was that k8s allow to use `subPath` in mounting (so not to override the whole directory), but that stops propagation of changes in ConfigMap to pods. Not an issue in the case of `/data`.
